tracer/build/_build/Build.ExplorationTests.cs

```diff
@@ -0,0 +1,301 @@
+                Name = "Paket",
+                GitRepositoryTag = "6.2.1",
+                PathToUnitTestProject = "tests/Paket.Tests",
+                TestsToIgnore = new[] { "Loading assembly metadata works" },
+                SupportedFrameworks = new[] { TargetFramework.NET461, TargetFramework.NETCOREAPP3_1 }
+            };
+        }
+
+        if (testName.Equals(ExplorationTestName.ILSpy))
+        {
+            return new ExplorationTestDescription()
+            {
+                GitRepositoryUrl = "https://github.com/icsharpcode/ILSpy.git",
+                Name = "ILSpy",
+                GitRepositoryTag = "6.2.1",
+                PathToUnitTestProject = "ICSharpCode.Decompiler.Tests",
+                TestsToIgnore = new[] { "ICSharpCode.Decompiler.Tests", "UseMc", "_net45", "ImplicitConversions", "ExplicitConversions", "ICSharpCode_Decompiler", "NewtonsoftJson_pcl_debug", "NRefactory_CSharp", "Random_TestCase_1" },
+                SupportedFrameworks = new[] { TargetFramework.NET461, TargetFramework.NETCOREAPP3_1 }
+            };
+        }
+
+        throw new NotImplementedException($"This exploration test '{testName}' is not supported");
+    }
+}
+
+partial class Build
+{
+    AbsolutePath ExplorationTestsDirectory => RootDirectory / "exploration-tests";
+
+    [Parameter("Indicates name of exploration test to run. If not specified, will run all tests sequentially.")]
+    readonly ExplorationTestName ExplorationTestName;
+
+    [Parameter("Indicates whether exploration tests should skip cloning. Default false.")]
+    readonly bool SkipClone;
+
+    [Parameter("Indicates whether build of exploration tests should be skipped. Useful for local development. Default false.")]
+    readonly bool SkipBuild;
+
+    [Parameter("Indicates whether exploration tests should run on latest repository commit. Useful if you want to update tested repositories to the latest tags. Default false.")]
+    readonly bool CloneLatest;
+
+    Target RunExplorationTests_Debugger
+        => _ => _
+               .Description("Run exploration tests for debugger.")
+               .After(Clean)
+               .DependsOn(PrepareMonitoringEnvironment_Debugger)
+               .Executes(() =>
+               {
+                   GitCloneAndRunUnitTests(MonitoringType.Debugger);
+               })
+               .Triggers(RunExplorationTestAssertions_Debugger)
+        ;
+
+    Target PrepareMonitoringEnvironment_Debugger
+        => _ => _
+           .Executes(() =>
+           {
+               Logger.Info($"Prepare environment variables for debugger.");
+               //TODO TBD
+           })
+        ;
+
+    Target RunExplorationTestAssertions_Debugger
+        => _ => _
+               .Description("Run exploration test assertions.")
+               .Unlisted()
+               .Executes(() =>
+               {
+                   Logger.Info($"Running assertions tests for debugger.");
+                   //TODO TBD
+               });
```

Comment:
I know these are just stubs, but I'm inclined to remove them for now - it seems likely they could easily be functions at the start and end of the  `RunExplorationTests_Debugger` target. We have a lot of targets already, so reducing the graph is useful I think 😄 Run `nuke --plan` and you'll see what I mean 😉

Reply:
Yeah, I've run `nuke --plan` once, got shocked :sweat_smile:
Replacing them with functions. :)